cmd/infra/azure/create.go

```diff
@@ -431,17 +444,26 @@ func createVirtualNetwork(ctx context.Context, subscriptionID string, resourceGr
 		},
 	}, nil)
 	if err != nil {
-		return "", "", "", fmt.Errorf("failed to create vnet: %w", err)
+		return armnetwork.VirtualNetworksClientCreateOrUpdateResponse{}, fmt.Errorf("failed to create vnet: %w", err)
 	}
 	vnet, err := vnetFuture.PollUntilDone(ctx, nil)
 	if err != nil {
-		return "", "", "", fmt.Errorf("failed to wait for vnet creation: %w", err)
+		return armnetwork.VirtualNetworksClientCreateOrUpdateResponse{}, fmt.Errorf("failed to wait for vnet creation: %w", err)
 	}
+
+	if vnet.ID == nil || vnet.Name == nil {
+		return armnetwork.VirtualNetworksClientCreateOrUpdateResponse{}, fmt.Errorf("created vnet has no ID or name")
+	}
+
 	if vnet.Properties.Subnets == nil || len(vnet.Properties.Subnets) < 1 {
-		return "", "", "", fmt.Errorf("created vnet has no subnets: %+v", vnet)
+		return armnetwork.VirtualNetworksClientCreateOrUpdateResponse{}, fmt.Errorf("created vnet has no subnets: %+v", vnet)
+	}
+
+	if vnet.Properties.Subnets[0].ID == nil || vnet.Properties.Subnets[0].Name == nil {
```

Comment:
I'm not sure if it will fail if the subnet fails to create; however, this function is only used for development/testing purposes. The check is primarily so I can trust I'm not going to nil pointer deference later - https://github.com/openshift/hypershift/blob/021003d7138a79f59165875ed5a71dfe6bc2e7b7/cmd/infra/azure/create.go#L209.